spark-indexer/src/main/java/org/uniprot/store/spark/indexer/genecentric/mapper/FastaToRelatedGeneCentricEntry.java

```diff
@@ -0,0 +1,78 @@
+package org.uniprot.store.spark.indexer.genecentric.mapper;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.uniprot.core.fasta.UniProtKBFasta;
+import org.uniprot.core.genecentric.GeneCentricEntry;
+import org.uniprot.core.genecentric.Protein;
+import org.uniprot.core.genecentric.impl.GeneCentricEntryBuilder;
+import org.uniprot.core.genecentric.impl.ProteinBuilder;
+import org.uniprot.core.parser.fasta.uniprot.UniProtKBFastaParser;
+import org.uniprot.core.uniprotkb.UniProtKBAccession;
+import org.uniprot.core.uniprotkb.impl.UniProtKBAccessionBuilder;
+
+import scala.Tuple2;
+
+/**
+ * @author lgonzales
+ * @since 21/10/2020
+ */
+public class FastaToRelatedGeneCentricEntry extends FastaToGeneCentricEntry {
+
+    private static final long serialVersionUID = -8317477346908862682L;
+
+    @Override
+    Tuple2<String, GeneCentricEntry> parseEntry(
```

Comment:
this method looks to be converting the string -> genecentricentry. is it poss to extract the logic into a common class, which both the spark+spring batch ways can use?

Reply:
If we plan on having a generic framework that tests the contents of the converted object are equal for spark+spring batch ways, then that's fine. I'm just thinking ahead perhaps ...

What do you think?